zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterFactory.java

```diff
@@ -332,10 +332,25 @@ private void loadFromFile() throws IOException {
       // previously created setting should turn this feature on here.
       setting.getOption().setRemote(true);
 
-      InterpreterSetting intpSetting = new InterpreterSetting(setting.id(), setting.getName(),
+      final InterpreterSetting intpSetting = new InterpreterSetting(setting.id(), setting.getName(),
           setting.getGroup(), setting.getInterpreterInfos(), setting.getProperties(),
           setting.getDependencies(), setting.getOption());
 
+      Thread t = new Thread() {
+        public void run() {
+          try {
+            loadInterpreterDependencies(intpSetting);
+          } catch (Exception e) {
+            logger.error(String.format("Error while downloading repos for interpreter group : %s," +
+                " go to interpreter setting page click on edit and save it again to make this " +
+                "interpreter work properly." ,
+                intpSetting.getGroup()), e);
+          }
+        }
+      };
+      t.start();
```

Comment:
Should this be synchronous?
If there are number of deps, the download might take a longer time. But the interpreter will be ready to take the paras.
If the code executed depends on these deps, it will fail I guess(until the time it is downloaded). And pass again after it is downloaded. Which will be kind of unexpected behaviour.